### ISSUE TITLE
1.6.0 not available in plugins.gradle.org/m2

### ISSUE THREAD

**Issue by bezda** [google/osdetector-gradle-plugin#15]
The version 1.6.0 of the plugin is not accessible

https://plugins.gradle.org/m2/gradle/plugin/com/google/gradle/osdetector-gradle-plugin/1.6.0/

osdetector-gradle-plugin-1.6.0.jar

```
<Error>
<Code>NoSuchKey</Code>
<Message>The specified key does not exist.</Message>
<Key>
gradle.plugin.com.google.gradle/osdetector-gradle-plugin/1.6.0/e074d3daa0ca0e5a0c6e2c98fcd9da6645ac520c8d091612ae58bdcde65d0585/osdetector-gradle-plugin-1.6.0.jar
</Key>
<RequestId>3C8CE86E331C7F5E</RequestId>
<HostId>
4KAux3R5OCNzgq4rDdQPaU6tU9amZxTb07yxbCtddy0wnL7XGbWItn5hCbq51oh5
</HostId>
</Error>
```

**Comment by tiainen** [comment]
I can confirm this. It is not possible to apply the plugin with a `plugins {}` block using version 1.6.0. This results in the following build failure:

    > Could not resolve all artifacts for configuration ':classpath'.
       > Could not find gradle.plugin.com.google.gradle:osdetector-gradle-plugin:1.6.0.
         Searched in the following locations:
           - https://plugins.gradle.org/m2/gradle/plugin/com/google/gradle/osdetector-gradle-plugin/1.6.0/osdetector-gradle-plugin-1.6.0.pom
           - https://plugins.gradle.org/m2/gradle/plugin/com/google/gradle/osdetector-gradle-plugin/1.6.0/osdetector-gradle-plugin-1.6.0.jar
         Required by:
             project : > com.google.osdetector:com.google.osdetector.gradle.plugin:1.6.0

**Comment by normano** [comment]
Resolved it putting this in settings.gradle

pluginManagement {
  repositories {
    mavenCentral()
    gradlePluginPortal()
  }

  resolutionStrategy {
    eachPlugin {

      if(requested.id.namespace == 'com.google' && requested.id.name == 'osdetector') {
        useModule 'com.google.gradle:osdetector-gradle-plugin:1.6.0'
      }
    }
  }
}

**Comment by aalmiray** [comment]
Please refer to https://docs.gradle.org/current/userguide/java_gradle_plugin.html and configure the `java-gradle-plugin` plugin in this build. The problem is caused by missing metadata that this plugin can generate.

**Comment by brunoborges** [comment]
Hey @zhangkun83, any chance to get this fixed?

**Comment by zhangkun83** [comment]
Checked the jar files of 1.4.0 and 1.6.0, didn't find anything different in their metadata. I added `java-gradle-plugin` and built a jar, and resulted in the same metadata. 

Found related issue on the forum:
https://discuss.gradle.org/t/error-nosuchkey-for-plugin-https-plugins-gradle-org-m2-org-standardout-gradle-eclipseconfig-1-1-1/29338

Although I doubt this issue is due to the build file, I will add `java-gradle-plugin` to the build file and do a new release.

**Comment by aalmiray** [comment]
The missing metadata is not supposed to be located inside the plugin JAR file but rather as a separate POM file that’s published alongside said JAR. This additional POM is generated by the `java-gradle-plugin` and published by the the `plugin-publish` plugin, like the plugin development guide explains.

**Comment by zhangkun83** [comment]
I have released 1.6.1. I successfully downloaded files under

https://plugins.gradle.org/m2/gradle/plugin/com/google/gradle/osdetector-gradle-plugin/1.6.1/

Looks like the issue is resolved. Please reopen it if it's not the case.

**Comment by aalmiray** [comment]
Thank you very much for looking at this issue and providing a fix! ! :-)
